### PR TITLE
[1LP][RFR] Replace usage of entity.check() with entity.ensure_checked. Part 1.

### PR DESCRIPTION
**Pull request by prichard77**
## Purpose or Intent

- __Updating tests and framework__ to use entity ensure_checked() instead of entity.check(). ensure_checked() is an enhancement to check() functionality that accounts for the checked state of an entity when selecting it.

### PRT Run

{{ pytest: --use-provider rhv42 --use-provider rhos13 --long-running cfme/tests/cloud_infra_common/test_policy_simulation.py::test_policy_simulation_ui cfme/tests/cloud/test_instance_power_control.py::test_power_on_or_off_multiple cfme/tests/cloud_infra_common/test_vm_instance_analysis.py::test_ssa_multiple_vms cfme/tests/infrastructure/test_host.py::test_multiple_host_good_creds }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/cloud/instance/__init__.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def power_control_from_cfme(self, *args, **kwargs):
                 raise ItemNotFound(
                     'Failed to find instance in table: {}'.format(self.name)
                 )
-            row.check()
+            row.ensure_checked()
```
Comment (prichard77) [inline cfme/cloud/instance/__init__.py]: I have only been able to test this by changing the surrounding code so line 321 would be executed. IT tested fine, but there is not a specific test in PRT that covers this update.

File: cfme/cloud/instance/__init__.py [hunk 2]
```diff
@@ -643,7 +643,8 @@ def prerequisite(self):
     def step(self, *args, **kwargs):
         # click the checkbox of every object in the filtered collection
         for entity in self.obj.all():
-            self.prerequisite_view.entities.get_entity(name=entity.name, surf_pages=True).check()
+            self.prerequisite_view.entities.get_entity(name=entity.name,
```
Comment (prichard77) [inline cfme/cloud/instance/__init__.py]: This is covered by cfme/tests/cloud_infra_common/test_policy_simulation.py::test_policy_simulation_ui and will be added to PRT.

File: cfme/cloud/provider/__init__.py [hunk 3]
```diff
@@ -266,7 +266,8 @@ class Edit(CFMENavigateStep):
     prerequisite = NavigateToSibling('All')
 
     def step(self, *args, **kwargs):
-        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).check()
+        self.prerequisite_view.entities.get_entity(name=self.obj.name,
+                                                   surf_pages=True).ensure_checked()
```
Comment (prichard77) [inline cfme/cloud/provider/__init__.py]: I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

File: cfme/cloud/provider/__init__.py [hunk 4]
```diff
@@ -289,7 +290,8 @@ class EditTags(CFMENavigateStep):
...
```
Comment (prichard77) [inline cfme/cloud/provider/__init__.py]: I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

File: cfme/common/provider.py [hunk 5]
```diff
@@ -744,7 +744,7 @@ def refresh_provider_relationships_ui(self, from_list_view=False, wait=0, delay=
         if from_list_view:
             view = navigate_to(self, 'All')
             entity = view.entities.get_entity(name=self.name, surf_pages=True)
-            entity.check()
+            entity.ensure_checked()
```
Comment (prichard77) [inline cfme/common/provider.py]: I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

File: cfme/tests/cloud/test_instance_power_control.py [hunk 6]
```diff
@@ -330,8 +330,8 @@ def _get_view_with_icons_checked():
         view = navigate_to(testing_instance.parent, 'All')
         view.toolbar.view_selector.select('Grid View')
         view.paginator.set_items_per_page(1000)
-        view.entities.get_entity(name=testing_instance.name).check()
-        view.entities.get_entity(name=testing_instance2.name).check()
+        view.entities.get_entity(name=testing_instance.name).ensure_checked()
+        view.entities.get_entity(name=testing_instance2.name).ensure_checked()
```
Comment (prichard77) [inline cfme/tests/cloud/test_instance_power_control.py]: This test will be part of PRT.

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 7]
```diff
@@ -971,7 +971,7 @@ def test_ssa_multiple_vms(ssa_multiple_vms, soft_assert, appliance, compare_linu
     view.toolbar.view_selector.select('List View')
     view.paginator.set_items_per_page(1000)
     for ssa_vm in ssa_multiple_vms:
-        view.entities.get_entity(name=ssa_vm.name, surf_pages=True).check()
+        view.entities.get_entity(name=ssa_vm.name, surf_pages=True).ensure_checked()
```
Comment (prichard77) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Tested and I am adding to PRT.

File: cfme/tests/infrastructure/test_host.py [hunk 8]
```diff
@@ -90,7 +90,7 @@ def navigate_and_select_quads(provider):
         view: the provider nodes view, quadicons already selected"""
     hosts_view = navigate_to(provider, 'ProviderNodes')
     assert hosts_view.is_displayed
-    [h.check() for h in hosts_view.entities.get_all()]
+    [h.ensure_checked() for h in hosts_view.entities.get_all()]
```
Comment (prichard77) [inline cfme/tests/infrastructure/test_host.py]: This has been tested with cfme/tests/infrastructure/test_host.py::test_multiple_host_good_creds and the test will be added to PRT.

**Comment by prichard77** [comment]
Something seems to be wrong with my PRT format. It's not running what I expect. Output doesn't tell me what is wrong , so I'll just keep trying until it works or I give up.

**Comment by prichard77** [comment]
I have updated the format of my PRT command a couple times. I'll get some assistance debugging this on Monday. Am marking RFR to possibly get this looked at while trying to figure out the PRT issue. I also added comments about the testing of each update.

**Pull request merged**